### ISSUE TITLE
Hitting API limit when tagging LDD repos

### ISSUE THREAD

**Issue by jordanpadams** [NASA-PDS/sumo-tools#1]
## 🐛 Describe the bug

When attempting to run the [prep_for_ldd_release.sh](https://github.com/NASA-PDS/operations/blob/main/bin/ldds/prep_for_ldd_release.sh) script, numerous builds fail as we hit the API access limit for Github.

## 📜 To Reproduce
1. Run prep_for_ldd_release.sh
2. See numerous repos fail their builds
3. Check the API rate limit for the pdsen-ci user:
```
$ curl -u "pdsen-ci:$GITHUB_TOKEN"  -H "Accept: application/vnd.github.v3+json"   https://api.github.com/rate_limit

{
  "resources": {
    "core": {
      "limit": 5000,
      "used": 4648,
      "remaining": 352,
      "reset": 1635783140
    },
...
  "rate": {
    "limit": 5000,
    "used": 4648,
    "remaining": 352,
    "reset": 1635783140
  }
}
```

## 🕵️ Expected behavior

All execute successfully

---

## ⚙️ Engineering Details
While checking the rate limit through the build, the first couple repos I tried to build synchronously (ldd-img, ldd-msn, etc) it looked like they only took ~250 requests for each repo. Once I tried building many at one, the number of requests shot up to >4000.

This issue is definitely somewhere in [pds-github-util](https://github.com/NASA-PDS/pds-github-util/) and how it is accessing the Github API. Need to figure out how to dampen the number of requests.

**Comment by jordanpadams** [comment]
most likely related to https://github.com/NASA-PDS/devops/issues/14

**Comment by jordanpadams** [comment]
Closed per 0834a175b34302c1b4190ffeb56f253e3314f210 and additional follow-on commits

**Comment by jordanpadams** [comment]
Down to 8 API hits per execution of sumo-tools
